File: qc-core/src/main/java/com/eager/questioncloud/core/domain/coupon/infrastructure/repository/CouponJpaRepository.java
package com.eager.questioncloud.core.domain.coupon.infrastructure.repository;

import com.eager.questioncloud.core.domain.coupon.infrastructure.entity.CouponEntity;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CouponJpaRepository extends JpaRepository<CouponEntity, Long> {
    Optional<CouponEntity> findByCode(String code);
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/coupon/infrastructure/repository/CouponJpaRepository.kt
package com.eager.questioncloud.core.domain.coupon.infrastructure.repository

import com.eager.questioncloud.core.domain.coupon.infrastructure.entity.CouponEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface CouponJpaRepository : JpaRepository<CouponEntity, Long> {
    fun findByCode(code: String): Optional<CouponEntity>
}
